### PR TITLE
Rb photogram industrial 2

### PR DESCRIPTION
**Pull request by hcontr4**
Git commits up to part 2.

### CODE REVIEW

**Review by rayguna — APPROVED** [review]
Looks good. The schema.db and the models files looks complete.

File: .vscode/settings.json
Comment (thierrychau) [inline .vscode/settings.json:1]: I'm inclined to say that this file does not need to be committed to your repo. In the future, you can either add it to your `.gitignore`, or just not stage it when you commit.

For this time, it's okay to leave.

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#  index_comments_on_photo_id  (photo_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (author_id => users.id)
+#  fk_rails_...  (photo_id => photos.id)
+#
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: "User", counter_cache: true
+  belongs_to :photo
```
Comment (thierrychau) [inline app/models/comment.rb]: Can you add a counter_cache?

File: app/models/follow_request.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+#  index_follow_requests_on_sender_id     (sender_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (recipient_id => users.id)
+#  fk_rails_...  (sender_id => users.id)
+#
+class FollowRequest < ApplicationRecord
+  belongs_to :recipient, class_name: "User"
+  belongs_to :sender, class_name: "User"
```
Comment (thierrychau) [inline app/models/follow_request.rb:24]: You could greatly benefit from having an `enum` for the `status`. Can you add this?

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+#  index_likes_on_photo_id  (photo_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (fan_id => users.id)
+#  fk_rails_...  (photo_id => photos.id)
+#
+class Like < ApplicationRecord
+  belongs_to :fan, class_name: "User", counter_cache: true
+  belongs_to :photo, counter_cache: true
```
Comment (thierrychau) [inline app/models/like.rb:23]: Can you add a validation to make sure a user is not able to like the same photo more than once?

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  has_many :own_photos, class_name: "Photo", foreign_key: "ownner_id"
+  has_many :comments, foreign_key: "author_id"
+
+  has_many :likes, foreign_key: :fan_id
+
+  has_many :sent_follow_requests, foreign_key: :sender_id, class_name: "FollowRequest"
```
Comment (thierrychau) [inline app/models/user.rb]: Can you add a `dependent: :destroy` option to take care of orphan records?

**Review by thierrychau — CHANGES_REQUESTED** [review]
Hi Herman, great job with implementing all these resources. I have a few minor comments, please see below. Also, did you create a sample_data task?

**Review by thierrychau — APPROVED** [review]
Looks great, good to merge. Thanks for the edits.

**Comment by MauritzKurt** [comment]
Looks complete, great job on the backend setup!

**Pull request merged**